apps/src/music/views/InstrumentGrid/index.tsx

```diff
@@ -262,6 +304,31 @@ const InstrumentGrid: React.FunctionComponent<Props> = ({
         event.stopPropagation();
         break;
       }
+      case 'ArrowLeft':
+      case 'ArrowRight':
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        event.preventDefault();
+        // Find all focusable cells in tab order
+        const focusableCells = Array.from(
+          document.querySelectorAll<HTMLElement>(`.${SHOWING}`)
+        );
```

Comment:
Would it be possible to avoid querying the React-rendered DOM like this?  This feels fragile compared to having a native React implementation.

Reply:
I just refactored these files to create and maintain a 2d array of refs. While this makes the arrow key navigation less math intensive, it's got its own quirks (some of the rows are hidden, so we end up with rows of empty refs). I opted to keep the empty refs and handle them in navigation instead of further refactoring the way the grid is rendered. Let me know if this better matches what you were expecting. I also refactored the child component EaseIntoView to eliminate the DOM query there as well. The description is updated (and the video new) for additional clarity.

Reply:
Looking at this now!

Reply:
Aha, I am able to repro on production, but it requires first changing the song to something like Strangers by Kenya Grace: the lowest note is D3, and there is no C3 listed, due to its key signature.

Reply:
Ohhhh thank you for this! Was able to repro and opened a fix forward here: https://github.com/code-dot-org/code-dot-org/pull/68929